src/list/list.ts

```diff
@@ -0,0 +1,229 @@
+  }
+
+  public remove(searchFunction: SearchFunction<T>): T | undefined {
+    if (!this.firstNode) {
+      return undefined;
+    }
+    let prevNode = this.memory.get(this.firstNode);
+    let node = this.memory.get(this.firstNode);
+    while (node) {
+      const { content, next } = node;
+      if (searchFunction(content)) {
+        prevNode.next = next;
+        this.memory.delete(prevNode.next);
+        return content;
+      }
+      prevNode = node;
+      node = this.memory.get(node.next);
+    }
+    return undefined;
+  }
+
+  public removeLast(): T | null {
+    if (!this.firstNode) {
+      return null;
+    }
+    let prevNode = this.memory.get(this.firstNode);
+    let node = this.memory.get(this.firstNode);
+    while (node) {
+      const { next } = node;
+      if (next === this.lastNode) {
+        const { content } = this.memory.get(next);
+        this.lastNode = prevNode.next;
+        node.next = null;
+        this.memory.delete(next);
+        this.length -= 1;
+        if (this.length === 1) {
+          this.lastNode = this.firstNode;
+        }
+        return content;
+      }
+      prevNode = this.memory.get(node.next);
+      node = this.memory.get(next);
+    }
+    return null;
+  }
+
+  public find(searchFunction: SearchFunction<T>, globalSearch = false): T | T[] {
+    if (!this.firstNode) {
+      return [];
+    }
+    const matchContent = [];
+    let node: Nodee<T> = this.memory.get(this.firstNode);
+    while (node) {
+      const { content, next } = node;
+      if (searchFunction(content)) {
+        if (globalSearch) {
+          matchContent.push(content);
+        } else {
+          return content;
+        }
+      }
+      node = this.memory.get(next);
+    }
+    return matchContent;
+  }
+
+  public forEach(callback: (content: T) => void): void {
+    if (this.firstNode) {
+      let node: Nodee<T> = this.memory.get(this.firstNode);
+      while (node) {
+        callback(node.content);
+        node = this.memory.get(node.next);
+      }
+    }
+  }
+
+  public map(callback: Function):List<T> {
+    const mapList = new List<T>();
+    if (!this.firstNode) {
+      return mapList;
+    }
+    let node: Nodee<T> = this.memory.get(this.firstNode);
+    let mappedElement: T;
+    while (node) {
+      if (node.next !== null) {
+        mappedElement = callback(node.content);
+        mapList.add(mappedElement);
+      }
+      node = this.memory.get(node.next);
+    }
+    return mapList;
+  }
+
+  public filter(filterFunction: SearchFunction<T>): List <T> {
+    const filterList = new List<T>();
+    if (!this.firstNode) {
+      return new List<T>();
+    }
+    let node: Nodee<T> = this.memory.get(this.firstNode);
+    while (node) {
+      if (filterFunction(node.content)) {
+        filterList.add(node.content);
+      }
+      node = this.memory.get(node.next);
+    }
+    return filterList;
+  }
+
+  public reducer(init: T, callback: (acc: T, currentVal: T) => T):T {
```

Comment:
first is the callback, and second argument is the initialValue.
Change name init to initialValue, the initialValue is not required to be type T, it could be whatever you want (use any)
The reduce function doesn't return T, it returns whatever you want

public reduce<R=any, S=any>(callback: (acc: R, currentValue: T) => R, initialValue: S): R